sktime/forecasting/tests/test_neuralforecast.py

```diff
@@ -225,29 +227,122 @@ def test_neural_forecast_with_auto_against_given_freq(model_class, freq) -> None
+
+    model.fit(y, fh=[1, 2, 3])
+    model_auto.fit(y, fh=[1, 2, 3])
+
+    pred = model.predict()
+    pred_auto = model_auto.predict()
+
+    # check prediction
+    pandas.testing.assert_series_equal(pred, pred_auto)
+
+
+@pytest.mark.parametrize(
+    "index",
+    [
+        pandas.RangeIndex(start=0, stop=20),
+        pandas.Index(range(20)),
+    ],
+)
+@pytest.mark.parametrize("model_class", [NeuralForecastLSTM, NeuralForecastRNN])
+@pytest.mark.skipif(
+    not run_test_for_class([NeuralForecastLSTM, NeuralForecastRNN]),
+    reason="run test only if softdeps are present and incrementally (if requested)",
+)
+def test_neural_forecast_with_auto_freq_on_missing_int_like(index, model_class) -> None:
+    """Test with freq set to 'auto' on int-like index with missing values."""
+
+    index = index.drop(random.choices(index, k=5))
+    y = pandas.Series(data=range(len(index)), index=index)
+
+    model = model_class(freq="auto", max_steps=1, trainer_kwargs={"logger": False})
 
-    # should fail to interpret auto freq
     with pytest.raises(
         ValueError,
-        match="could not interpret freq, try passing freq in model initialization",
+        match="(could not interpret freq).*(use a valid integer offset in index)",
     ):
-        # define model
-        model = model_class(freq="auto", max_steps=5, trainer_kwargs={"logger": False})
+        model.fit(y, fh=[1, 2, 3])
 
-        # attempt train
-        model.fit(y, fh=[1, 2, 3, 4])
 
-    # should work with freq passed as param
-    model = model_class(freq="W", max_steps=5, trainer_kwargs={"logger": False})
+@pytest.mark.parametrize(
+    "index",
+    [
+        pandas.date_range(start="2024-01-01", periods=10),
+        pandas.period_range(start="2024-01-01", periods=10),
+    ],
+)
+@pytest.mark.parametrize("model_class", [NeuralForecastLSTM, NeuralForecastRNN])
+@pytest.mark.skipif(
+    not run_test_for_class([NeuralForecastLSTM, NeuralForecastRNN]),
+    reason="run test only if softdeps are present and incrementally (if requested)",
+)
+def test_neural_forecast_with_auto_freq_on_date_like(index, model_class) -> None:
+    """Test with freq set to 'auto' on date-like index."""
 
-    # attempt train
-    model.fit(y, fh=[1, 2, 3, 4])
+    y = pandas.Series(data=range(len(index)), index=index)
+
+    model = model_class(freq="D", max_steps=1, trainer_kwargs={"logger": False})
+    model_auto = model_class(freq="auto", max_steps=1, trainer_kwargs={"logger": False})
+
+    model.fit(y, fh=[1, 2, 3])
+    model_auto.fit(y, fh=[1, 2, 3])
+
+    pred = model.predict()
+    pred_auto = model_auto.predict()
+
+    # check prediction
+    pandas.testing.assert_series_equal(pred, pred_auto)
+
+
+@pytest.mark.parametrize(
+    "index",
+    [
+        # DatetimeIndex
+        pandas.date_range(start="2024-01-01", periods=5),
+        pandas.to_datetime(
+            ["2000-01-01", "2000-01-02", "2000-01-03", "2000-01-04", "2000-01-05"]
+        ),
```

Comment:
Both of these tests seems to be failing in CI.

Reply:
It was working fine before this commit https://github.com/sktime/sktime/pull/6237/commits/83aa7ca43217234d0131db0487646e25e638d168 where I merge `main` branch into the current branch but suddenly after merging, `ForecastingHorizon` is able to infer freq from both `PeriodIndex` and `DatetimeIndex`. Has there been some changes to `ForecastingHorizon` directly or indirectly in main?

Reply:
I am not aware of, may be @fkiraly knows. There are some works on `fh` going on (mainly by @benHeid) but I don't think those are merged yet.

Were you up to date with `sktime==0.28` last time the tests worked? Can these be affected by pandas 2.2 versions?

Reply:
Oh! actually what happens is that, to create missing date-like series, I am dropping some indexes using this line of code
```py
index = index.drop(random.choices(index, k=2))
```
since the behavior is random, if it drop the last 2 or the first 2 indexes, the complete series tends to be without any missing value and as a result no error is raised
I will handle this, if you have an opinion, let me know here